application/client.plugins/projects/serial/src/lib/services/service.ts

```diff
@@ -170,28 +171,62 @@ export class Service extends Toolkit.APluginService {
             command: EHostCommands.spyStart,
             options: options,
         }, this.session).catch((error: Error) => {
-            this._logger.error(error);
+            throw(error);
         });
     }
 
     public stopSpy(options: IOptions[]): Promise<any> {
+        if (options.length > 0) {
+            return this.api.getIPC().requestToHost({
+                stream: this.session,
+                command: EHostCommands.spyStop,
+                options: options,
+            }, this.session).catch((error: Error) => {
+                throw(error);
+            });
+        }
+        return new Promise((resolve) => {
+            resolve();
+        });
```

Comment:
if you need just resolve promise with returning, better to use:
```
return Promise.resolve();
```
because it's more clean